src/qibo/base/hamiltonians.py

```diff
@@ -187,6 +211,179 @@ def __matmul__(self, o):
                                              "implemented.".format(type(o)))
 
 
+class SymbolicHamiltonian:
+    """Parses symbolic Hamiltonians defined using ``sympy``.
+
+    This class should not be used by users.
```

Comment:
Shall we rename this class as `_SymbolicHamiltonian`?

Reply:
It is not appearing in the docs anyway, but I renamed it using the `_` since it is not supposed to be used by the user.